packages/browser-wallet-api/src/wallet-api.ts

```diff
@@ -1,27 +1,21 @@
-import {
-    InjectedMessageHandler,
-    createEventTypeFilter,
-    EventType,
-    MessageType,
-} from '@concordium/browser-wallet-message-hub';
+import { InjectedMessageHandler, createEventTypeFilter, MessageType } from '@concordium/browser-wallet-message-hub';
 import { AccountTransactionPayload, AccountTransactionSignature, AccountTransactionType } from '@concordium/web-sdk';
-import { WalletApi as IWalletApi, WalletEventHandler } from '@concordium/browser-wallet-api-helpers';
+import { WalletApi as IWalletApi, EventType } from '@concordium/browser-wallet-api-helpers';
+import EventEmitter from 'events';
 import { stringify } from './util';
 
-class WalletApi implements IWalletApi {
+class WalletApi extends EventEmitter implements IWalletApi {
     private messageHandler = new InjectedMessageHandler();
 
     private connected = false;
 
-    private eventHandlerMap: Map<EventType, WalletEventHandler[]> = new Map();
-
     constructor() {
-        // set up event listeners
-        this.handleEvent(EventType.ChangeAccount);
-    }
+        super();
 
-    public addChangeAccountListener(handler: WalletEventHandler<string>) {
-        this.addEventListener(EventType.ChangeAccount, handler);
+        // Set up message handlers to emit events.
+        Object.values(EventType).forEach((eventType) => {
+            this.handleEvent(eventType);
+        });
```

Comment:
```suggestion
        Object.values(EventType).forEach((eventType) => 
            this.handleEvent(eventType)
        );
```

Reply:
Could also simply: `Object.values(EventType).forEach(this.handleEvent.bind(this))`

Reply:
:vomiting_face:

Reply:
Søren wins.